### ISSUE TITLE
On Ractive.extend(), warn if data contains non-primitives

### ISSUE THREAD

**Issue by Rich-Harris** [ractivejs/ractive#1803]
I recall discussing this recently, but I don't think an issue was created for it

**Comment by martypdx** [comment]
Isn't it covered [here](https://github.com/ractivejs/ractive/blob/dev/src/Ractive/config/custom/data.js#L9)?

**Comment by Rich-Harris** [comment]
No, I mean this sort of thing:

```js
X = Ractive.extend({
  data: {
    answer: 42,     // fine
    emptyArray: [], // not fine, might not be empty for long
    obj: {}         // careful!
  }
});

r1 = new X();
r2 = new X();

r1.set( 'obj.foo', 'bar' );
r2.push( 'emptyArray', 'trololol' );

r2.get( 'obj.foo' ); // 'bar'
r1.get( 'emptyArray' ); // 'trololol'
```

I think that on `Ractive.extend()`, we should check all the properties of `data` and check that they're primitives. If they're not, we should show people how to use data functions to avoid that kind of pollution.

**Comment by Rich-Harris** [comment]
closed via #1812
